src/Sema.zig

```diff
@@ -8599,8 +8600,22 @@ fn zirIntFromEnum(sema: *Sema, block: *Block, inst: Zir.Inst.Index) CompileError
     };
     const enum_tag_ty = sema.typeOf(enum_tag);
 
+    // TODO: use correct solution
+    // https://github.com/ziglang/zig/issues/15909
+    if (enum_tag_ty.enumFieldCount(mod) == 0 and !enum_tag_ty.isNonexhaustiveEnum(mod)) {
+        return sema.fail(block, operand_src, "cannot use @intFromEnum on empty enum '{}'", .{
+            enum_tag_ty.fmt(mod),
+        });
+    }
+
     const int_tag_ty = enum_tag_ty.intTagType(mod);
 
+    if (try sema.resolveValue(operand)) |resolved_op| {
```

Comment:
Replace the other use of `resolveValue` with `resolveDefinedValue` instead.

Reply:
Is this not exactly what you were telling me not to do?
```zig
if (val.isUndef(mod)) {
    return sema.failWithUseOfUndef(block, src);
}
```

Reply:
No, I was wrong. But you should still put that `isUndef` in the other `resolveValue` branch.

Reply:
```zig
if (try sema.resolveValue(enum_tag)) |enum_tag_val| {
    if (enum_tag_val.isUndef(mod)) {
        return mod.undefRef(int_tag_ty);
    }

    const val = try enum_tag_val.intFromEnum(enum_tag_ty, mod);
    return Air.internedToRef(val.toIntern());
}
```?

Reply:
Yes something like that, or you could make `Value.intFromEnum` handle `.undef`.

Either way there is still some other bug that I don't have time to look into now but it's in the comment I made in the issue.

Reply:
I pushed this, so resolving.